internal/cmd/compilecheck/go.mod

```diff
@@ -6,8 +6,7 @@ replace go.mongodb.org/mongo-driver => ../../../
 
 // Note that the Go driver version is replaced with the local Go driver code by
 // the replace directive above.
-//require go.mongodb.org/mongo-driver
-require go.mongodb.org/mongo-driver v1.11.7
+//require go.mongodb.org/mongo-driver/v2 v2.0.0
```

Comment:
TODO: before merge uncomment and update with alpha

Reply:
This is ready: https://github.com/mongodb/mongo-go-driver/actions/runs/10098917125

Reply:
On further investigation, I think we will have to wait for a release that contains `v2` in the import path in order for the compile check to work. I suggest releasing alpha2 after this is merged, and then we can create a followup PR to fix the compile check.

@matthewdale Do you have thoughts?

Reply:
The same goes for all tests in internal/cmd that are modules.

Reply:
Doing this in a follow-up PR seems fine.